static/js/settings.js

```diff
@@ -40,6 +41,74 @@ $("body").ready(() => {
     });
 });
 
+// This function replaces get_language_list_for_template
+// function used previously in the backend for getting data
+// in the desired form for the language list modal used the frontend.
+function get_language_list_columns(default_language) {
+    const language_list = [];
+
+    // Only render languages with percentage transaltion >= 5%
+    for (const language of page_params.language_list) {
+        if (language.percent_translated === undefined || language.percent_translated >= 5) {
+            language_list.push({
+                code: language.code,
+                locale: language.locale,
+                name: language.name,
+                percent_translated: language.percent_translated,
+            });
+        }
+    }
+
+    const formatted_list = [];
+    const language_len = language_list.length;
+    const firsts_end = Math.floor(language_len / 2) + (language_len % 2);
+    const firsts = _.range(0, firsts_end);
+    const seconds = _.range(firsts_end, language_len);
+
+    const longest_zip = [];
+
+    // Create a zip (itertool.zip_longest in python)
+    for (const value of firsts) {
+        if (seconds[value] !== undefined) {
+            longest_zip.push([value, seconds[value]]);
+        }
+    }
+
+    for (const row of longest_zip) {
+        const item = [];
+        const zip_row = [
+            ["first", row[0]],
+            ["second", row[1]],
+        ];
+        for (const zip_value of zip_row) {
+            if (zip_value) {
+                const lang = language_list[zip_value[1]];
+                const name = lang.name;
+                let percent = name;
+                if (lang.percent_translated !== undefined) {
+                    percent = name + " (" + lang.percent_translated + "%)";
+                }
```

Comment:
This function needs work on variable names; `percent` should not be a string.

Reply:
This is something derived from already used keys. Maybe `name_with_percent` is a better choice since it is actually that.

Reply:
Yeah, this is great.